### DISCUSSION TITLE
Difference between DragStarting vs DragItemsStarting and Drop vs DropCompleted events

### DISCUSSION THREAD

**Discussion by gautambjain** [microsoft/microsoft-ui-xaml#7834] [Q&A]
I am new to WinUI. Can't figure out the difference between DragStarting/DragItemsStarting . And also between Drop/DropCompleted events. What should I use what? The controls I am using are ListView and TreeView.

**Comment by castorix** [comment]
Some MSDN articles explain Drag&Drop, like : [Windows 10 - Modern Drag and Drop for Windows Universal Applications](https://learn.microsoft.com/en-us/archive/msdn-magazine/2015/august/windows-10-modern-drag-and-drop-for-windows-universal-applications)

**Reply by gautambjain** [reply]
Thanks a lot. This link is useful. However, in my case only Drop is called on target. DropCompleted is not called on source. Is there any additional settings required (like setting any property on DragEventArgs).

I have  treeview and listview. Both have drop and dropcompleted events implemented. Plus candragitems, allowdrop properties are true.
